File: MCQUI/src/app/models/question.ts
import { Answer } from './answer';

export interface Question {
  questionId: number;          // [Key]
  questionText: string;        // [Required], [MaxLength(2000)]
  allowMultiple: boolean;      // 0/1 in DB, boolean in API/TS
  category?: string | null;    // [MaxLength(200)]
  subCategory?: string | null; // [MaxLength(200)]
  difficulty?: string | null;  // [MaxLength(50)]
  explanation?: string | null; // [MaxLength(4000)]
  createdAt: string;           // DateTime from API (usually ISO string)
  answers: Answer[];           // ICollection<Answer> -> Answer[]
  IsAnswerCorrect: boolean;
}

File: MCQ-BE/Models/Question.cs
using System.ComponentModel.DataAnnotations;

namespace MCQ.Models
{

    public class Question
    {
        [Key]
        public int QuestionId { get; set; }

        [Required]
        [MaxLength(2000)]
        public string QuestionText { get; set; } = null!;

        // 0 = Single correct, 1 = Multiple correct
        public bool AllowMultiple { get; set; } = false;

        [MaxLength(200)]
        public string? Category { get; set; }

        [MaxLength(200)]
        public string? SubCategory { get; set; }

        [MaxLength(50)]
        public string? Difficulty { get; set; }

        [MaxLength(4000)]
        public string? Explanation { get; set; }

        public DateTime CreatedAt { get; set; } // set by DB default (SYSUTCDATETIME)

        // Navigation: 1 Question -> many Answers
        public ICollection<Answer> Answers { get; set; } = new List<Answer>();
        public bool IsAnswerCorrect { get; set; } = false;
    }

}
